### PR TITLE
dev-console: revamp the landing page

### PR DESCRIPTION
**Pull request by deshmukhmayur**
# Closes 2133

# Explain the feature/fix

- Complete revamp of the developer console landing page
- Also refreshed some UI elements in the sidebar and header bar

## Does this PR introduce a breaking change

No.

## Screenshots

<details>
<summary>View Screenshots</summary>

| Page | Screenshot |
|---|---|
| Landing Page | Screenshot 2022-11-03 at 16-45-30 Developer Console |
| Sidebar and Header | Screenshot 2022-11-03 at 4 48 13 PM |

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Was this feature demoed and the design review approved?

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
 	// List of extensions which should be recommended for users of this workspace.
 	"recommendations": [
 		"bierner.lit-html",
```
Comment (akhilmhdh) [inline .vscode/extensions.json:7]: Is the vscode workspace files needed in commit

File: packages/developer-console-spa/src/common/components/NewProjectForm/index.tsx [hunk 2]
```diff
@@ -1,79 +1,85 @@
...
     mode: 'onBlur',
-    resolver: yupResolver( appSchema ),
+    resolver: yupResolver( projectSchema ),
     defaultValues: {
       name: '',
       description: '',
     },
   });
 
   useEffect( () => {
```
Comment (akhilmhdh) [inline packages/developer-console-spa/src/common/components/NewProjectForm/index.tsx:34]: You dont need a useEffect for this. 

You can use a derived state, that is just use `const isOpen = searchParams.get('new') === 'true' `

And use the hook provided by react router to get query params

**Review by akhilmhdh — CHANGES_REQUESTED** [review]

**Comment by akhilmhdh** [comment]
Also two more points wanna add here.

1. You can avoid wrapping every function with useCallback as its unnecessary overhead.
2. I would recommend switching to something like urql especially for dev console

File: packages/developer-console-spa/src/common/components/NewProjectForm/index.tsx (hunk 2 above)
Comment (deshmukhmayur) [reply packages/developer-console-spa/src/common/components/NewProjectForm/index.tsx]: Changed the `isOpen` to normal variable.

Btw, the `useSearchParams` hook is in react-router v6. This is not upgraded yet. Will do that in the next pr update.

File: .vscode/extensions.json (hunk 1 above)
Comment (deshmukhmayur) [reply .vscode/extensions.json]: Removed

**Review by akhilmhdh — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Pull request merged**